File: nysol_python/mcmd/methods/convTex/db/smp/func_sqsum.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='sqsum'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
id,v1,v2,v3
1,1,2,3
2,-5,2,1
3,1,,3
4,,,
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
'''
script['sh_code']='''
mcal c='sqsum(${v1},${v2},${v3})' a=rsl i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcal(c='sqsum(${v1},${v2},${v3})', a='rsl', i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='ワイルドカードを利用した例'
script['comment']='''
``v`` から始まる項目( ``v1,v2,v3`` )をワイルドカード「 ``v*`` 」によって指定している。
'''
script['sh_code']='''
mcal c='sqsum(${v*})' a=rsl i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mcal(c='sqsum(${v*})', a='rsl', i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='-nを利用した例'
script['comment']='''
``v2`` にNULL値を含む ``id=3`` の行の結果もNULLとなる。
'''
script['sh_code']='''
mcal c='sqsum(${v1},${v2},${v3},"-n")' a=rsl i=dat1.csv o=rsl3.csv
'''
script['py_code']='''
nm.mcal(c='sqsum(${v1},${v2},${v3},"-n")', a='rsl', i="dat1.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

File: olddoc/mcmd/jp/examples/func_sqsum.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
id,v1,v2,v3
1,1,2,3
2,-5,2,1
3,1,,3
4,,,
EOF
)}

############## 例1
title="基本例"
comment=<<'EOF'
EOF
scp=<<'EOF'
more dat1.csv
mcal c='sqsum(${v1},${v2},${v3})' a=rsl i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## 例2
title="ワイルドカードを利用した例"
comment=<<'EOF'
\verb|v|から始まる項目(\verb|v1,v2,v3|)をワイルドカード「\verb|v*|」によって指定している。
EOF
scp=<<'EOF'
mcal c='sqsum(${v*})' a=rsl i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)

############## 例3
title="-nを利用した例"
comment=<<'EOF'
\verb|v2|にNULL値を含む\verb|id=3|の行の結果もNULLとなる。
EOF
scp=<<'EOF'
mcal c='sqsum(${v1},${v2},${v3},"-n")' a=rsl i=dat1.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)

